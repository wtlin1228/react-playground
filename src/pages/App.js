import React, { useState } from 'react';

const App = () => {
  const [todoList, setTodoList] = useState(['Buy Milk', 'Write tutorial']);
  const [inputValue, setInputValue] = useState('');

  const handleSubmit = () => {
    setTodoList([...todoList, inputValue]);
    setInputValue('');
  };

  return (
    <div className='container'>
      <div className='row'>
        <div className='col-md-6'>
          <h2>Add Todo</h2>
        </div>
      </div>
      <form onSubmit={handleSubmit}>
        <div className='row'>
          <div className='col-md-6'>
            <input
              className='form-control'
              data-testid='todo-input'
              placeholder='Enter a task'
              type='text'
              onChange={e => setInputValue(e.target.value)}
              value={inputValue}
            />
          </div>
        </div>
        <div className='row'>
          <div className='col-md-6'>
            <button
              className='btn btn-primary'
              data-testid='add-task'
              type='submit'
            >
              Add Task
            </button>
          </div>
        </div>
      </form>
      <div className='row todo-list'>
        <div className='col-md-6'>
          <h3>Lists</h3>
          <ul data-testid='todos-ul'>
            {todoList.map((item, index) => (
              <li key={index}>
                <div>
                  {item}
                  <button className='btn btn-danger'>X</button>
                </div>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </div>
  );
};

export default App;
