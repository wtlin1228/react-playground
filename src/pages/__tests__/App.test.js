import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import App from '../App';

afterEach(cleanup);

it('matches snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

it('displays default todo items', () => {
  const { getByTestId } = render(<App />);
  const todoList = getByTestId('todos-ul');
  expect(todoList.children.length).toBe(2);
});

it('allows input', () => {
  const { getByTestId } = render(<App />);
  const todoInput = getByTestId('todo-input');
  const item = 'Learn React';
  todoInput.value = item;
  fireEvent.change(todoInput);
  expect(todoInput.value).toBe(item);
});

it('adds a new todo item', () => {
  const { getByTestId, getByText } = render(<App />);
  const todoInput = getByTestId('todo-input');
  const addTodoButton = getByText('Add Task');
  const todoList = getByTestId('todos-ul');
  const item = 'Learn React';
  todoInput.value = item;
  fireEvent.change(todoInput);
  fireEvent.click(addTodoButton);
  expect(todoList.children.length).toBe(3);
});
