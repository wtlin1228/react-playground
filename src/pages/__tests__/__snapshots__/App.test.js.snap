// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <h2>
          Add Todo
        </h2>
      </div>
    </div>
    <form>
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <input
            class="form-control"
            data-testid="todo-input"
            placeholder="Enter a task"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <button
            class="btn btn-primary"
            data-testid="add-task"
            type="submit"
          >
            Add Task
          </button>
        </div>
      </div>
    </form>
    <div
      class="row todo-list"
    >
      <div
        class="col-md-6"
      >
        <h3>
          Lists
        </h3>
        <ul
          data-testid="todos-ul"
        >
          <li>
            <div>
              Buy Milk
              <button
                class="btn btn-danger"
              >
                X
              </button>
            </div>
          </li>
          <li>
            <div>
              Write tutorial
              <button
                class="btn btn-danger"
              >
                X
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
